x/dex/keeper/contract.go

```diff
@@ -50,6 +50,18 @@ func (k Keeper) GetContract(ctx sdk.Context, contractAddr string) (types.Contrac
 	return res, nil
 }
 
+func (k Keeper) GetContractGasLimit(ctx sdk.Context, contractAddr []byte) (uint64, error) {
+	bech32ContractAddr := sdk.AccAddress(contractAddr).String()
+	contract, err := k.GetContract(ctx, bech32ContractAddr)
+	if err != nil {
+		return 0, err
+	}
+	rentBalance := contract.RentBalance
+	gasPrice := k.GetParams(ctx).SudoCallGasPrice
+	gasDec := sdk.NewDec(int64(rentBalance)).Quo(gasPrice)
```

Comment:
is it possible for `gasPrice` to be zero? If so we should have zero div protection here.

Reply:
good point, updated